.travis.yml

```diff
@@ -1,25 +1,52 @@
 language: python
 
+sudo: false
 
 env:
-    - CONDA="python=2.7"
-    - CONDA="python=3.4"
-    - CONDA="python=3.5"
-    - CONDA="python=3.6"
+  global:
+    - secure: "DdUE82IkNYo317595sy4H6z3GQoaVUTAAUDOOoJATa199ERxSQy8VlsMKELdRTrBEt76kSBJdjjJ7YSiOVSMt9qao9XWOWYjtfdX0KpbkmfiHMH6fmQnckNndMbPY8wa42fFusIWOxZL/jCay1Gq0GppjoHdSoaLGIk3hLkYE1U="
+
+addons:
+  apt:
+    packages: octave
+
+matrix:
+  fast_finish: true
+  include:
+  - python: 2.7
+    env: TEST_TARGET=default NUMPY=1.11
+  - python: 2.7
+    env: TEST_TARGET=default NUMPY=1.12
+  - python: 3.6
+    env: TEST_TARGET=default NUMPY=1.11
+  - python: 3.6
+    env: TEST_TARGET=default NUMPY=1.12
+  - python: 3.6
+    env: TEST_TARGET=publish NUMPY=1.12
 
 before_install:
-    - sudo apt-add-repository -y ppa:picaso/octave
-    - sudo apt-get update -qq
-    - sudo apt-get install -qq octave liboctave-dev
     - wget http://bit.ly/miniconda -O miniconda.sh
     - bash miniconda.sh -b -p $HOME/miniconda
     - export PATH="$HOME/miniconda/bin:$PATH"
-    - conda update --yes conda
-    # Install numpy and scipy with conda for faster tests.
-    - travis_retry conda create --yes -n test $CONDA pip numpy scipy
-    - source activate test
-    - travis_retry pip install -r requirements-dev.txt
-    - conda info -a
+    - conda update conda --yes
+    - conda config --add channels conda-forge --force
+    - conda create --yes -n TEST python=$TRAVIS_PYTHON_VERSION --file requirements-dev.txt
+    - source activate TEST
+    # Install after to ensure it will be downgraded when testing an older version.
+    - conda install --yes --quiet numpy=$NUMPY
+    - if [[ "$TRAVIS_PYTHON_VERSION" == "3.6" ]]; then
+        conda install --yes doctr ;
```

Comment:
`doctr` is very easy to use and reduces the bolierplate for auto-publishing significantly. Let me know if you have any question in case you want to use it in another project.

The docs are now available at:  https://ocefpaf.github.io/python-gsw it would be nice to get them uploaded to this org instead. I will open an issue for that.